### PR TITLE
Added upsert field to FileOptions

### LINKED ISSUE #7: Support upload with upsert
# Feature request

## Is your feature request related to a problem? Please describe.

I want to upload/update profile images. They are named by the user id. If the file already exists, an upload fails, so an update is necessary. But an update fails if the file doesn't exist yet. As stated in the [docs](https://supabase.io/docs/reference/javascript/storage-from-upload#upload-file) there is an option to enable upsert for an upload.

## Describe the solution you'd like

Add `upsert` option to `FileOptions`

## Additional context

When using `upload` when the file already exists, the following error is thrown: 
`duplicate key value violates unique constraint "bucketid_objname"`

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

Adds an `upsert` option to `upload` method

Closes #7 

## What is the current behavior?

There is no upsert option

## What is the new behavior?

Users can upsert files to their Supabase storage

## Other context

Tested locally and was able to successfully perform a upsert request.

### CODE REVIEW

File: lib/src/fetch.dart [hunk 1]
```diff
@@ -123,7 +123,8 @@ class Fetch {
       final request = http.MultipartRequest(method, Uri.parse(url))
         ..headers.addAll(headers)
         ..files.add(multipartFile)
-        ..fields['cacheControl'] = fileOptions.cacheControl;
+        ..fields['cacheControl'] = fileOptions.cacheControl
```
Comment (dshukertjr) [inline lib/src/fetch.dart:126]: Just discovered that this line needs to be `headers` as well while working on this PR. Will submit a separate PR to fix this once this is merged.

**Review by phamhieu — APPROVED** [review]
lgtm! thanks @dshukertjr
